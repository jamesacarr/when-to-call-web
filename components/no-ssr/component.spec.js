import React from 'react';
import { shallow } from 'enzyme';
import NoSSR from './component';

describe('</NoSSR>', () => {
  it('renders empty when canRender false', () => {
    const wrapper = shallow(<NoSSR><div>Test</div></NoSSR>);
    wrapper.setState({ canRender: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders renders children when canRender true', () => {
    const wrapper = shallow(<NoSSR><div>Test</div></NoSSR>);
    wrapper.setState({ canRender: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('sets canRender to false initially', () => {
    const wrapper = shallow(<NoSSR><div>Test</div></NoSSR>, { disableLifecycleMethods: true });
    expect(wrapper.state('canRender')).toEqual(false);
  });

  it('sets canRender to true after componentDidMount', () => {
    const wrapper = shallow(<NoSSR><div>Test</div></NoSSR>);
    expect(wrapper.state('canRender')).toEqual(true);
  });
});
