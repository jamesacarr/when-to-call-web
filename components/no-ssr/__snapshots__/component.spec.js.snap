// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</NoSSR> renders empty when canRender false 1`] = `""`;

exports[`</NoSSR> renders renders children when canRender true 1`] = `
<div>
  Test
</div>
`;
